game-backend/src/game-engine/intent/ConsumeIntent.test.ts

```diff
@@ -0,0 +1,90 @@
+import ConsumeIntent, { FOOD_REGENERATION_VALUE } from './ConsumeIntent';
+import StateBuilder from '../../test-helper/StateBuilder';
+import { State } from '../../../../darwin-types/State';
+import { UserContext } from '../../../../darwin-types/UserContext';
+import { Unit, MAX_HEALTH } from '../../../../darwin-types/game-objects/Unit';
+import { Food } from '../../../../darwin-types/game-objects/Food';
+
+describe('ConsumeIntent', () => {
+  const unitId1 = 'UNIT_ID_1';
+  const unitId2 = 'UNIT_ID_2';
+  const unitId3 = 'UNIT_ID_3';
+  const foodId1 = 'FOOD_ID_1';
+  const foodId2 = 'FOOD_ID_2';
+
+  let state: State;
+  let userContext1: UserContext;
+  let userContext2: UserContext;
+  let userContext3: UserContext;
+
+  let originalUnit1: Unit;
+  let originalUnit2: Unit;
+  let originalUnit3: Unit;
+
+  beforeEach(() => {
+    state = StateBuilder.buildState()
+      .addUnit({ id: unitId1, x: 1, y: 1, health: 65 })
+      .addFood({ id: foodId1, x: 1, y: 1 })
+      .addUnit({ id: unitId2, x: 2, y: 2, health: 90 })
+      .addFood({ id: foodId2, x: 2, y: 2 })
+      .addUnit({ id: unitId3, x: 5, y: 5, health: 20 })
+      .build();
+
+    originalUnit1 = state.objectMap[unitId1] as Unit;
+    originalUnit2 = state.objectMap[unitId2] as Unit;
+    originalUnit3 = state.objectMap[unitId3] as Unit;
+
+    userContext1 = {
+      unitId: unitId1,
+    };
+    userContext2 = {
+      unitId: unitId2,
+    };
+    userContext3 = {
+      unitId: unitId3,
+    };
+  });
+
+  it('handles consume properly', () => {
+    const intent = new ConsumeIntent();
+
+    const newState = intent.execute(state, userContext1);
+    const unit1 = newState.objectMap[unitId1] as Unit;
```

Comment:
What about extracting this to a common helper logic as this will be used all over the codebase.
e.g. `getFood(state, id): Food`